lib/logger/lib/logger/app.ex

```diff
@@ -34,6 +34,12 @@ defmodule Logger.App do
     primary_config = :logger.get_primary_config()
     :ok = :logger.set_primary_config(:level, default_level())
 
+    :ok =
+      Application.get_env(:logger, :metadata, [])
+      |> Map.new()
+      |> then(&Map.merge(primary_config.metadata, &1))
+      |> then(&:logger.set_primary_config(:metadata, &1))
```

Comment:
```suggestion
    with [_ | _] = metadata <- Application.fetch_env!(:logger, :metadata) do
      :ok = :logger.set_primary_config(:metadata, Enum.into(metadata, primary_config.metadata))
    end
```

So we don't bother setting up a metadata if nothing changed?

Reply:
Really nice. Done.